image/helpers.go

```diff
@@ -108,16 +109,28 @@ func StoreAssertionFetcher(sto Store, dlOpts *DownloadOptions, db *asserts.Datab
 }
 
 // FetchAndCheckSnapAssertions fetches and cross checks the snap assertions matching the given snap file using the provided asserts.Fetcher and assertion database.
-func FetchAndCheckSnapAssertions(snapPath string, info *snap.Info, f asserts.Fetcher, db asserts.RODatabase) error {
+func FetchAndCheckSnapAssertions(snapPath string, info *snap.Info, f asserts.Fetcher, db asserts.RODatabase) (*asserts.SnapDeclaration, error) {
 	sha3_384, size, err := asserts.SnapFileSHA3_384(snapPath)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
+	// this assumes series "16"
 	if err := snapasserts.FetchSnapAssertions(f, sha3_384); err != nil {
-		return fmt.Errorf("cannot fetch snap signatures/assertions: %v", err)
+		return nil, fmt.Errorf("cannot fetch snap signatures/assertions: %v", err)
 	}
 
 	// cross checks
-	return snapasserts.CrossCheck(info.Name(), sha3_384, size, &info.SideInfo, db)
+	if err := snapasserts.CrossCheck(info.Name(), sha3_384, size, &info.SideInfo, db); err != nil {
+		return nil, err
+	}
+
+	a, err := db.Find(asserts.SnapDeclarationType, map[string]string{
+		"series":  release.Series,
+		"snap-id": info.SnapID,
+	})
```

Comment:
I'm not sure how various checks here operate, please bear with me...
 - we hash the actual .snap file
 - we fetch and check (presumably that the signature itself is correct) all the assertions affecting the snap with the hash we've got
 - we cross-check that the snap name and snap hash are correct and match side info

...and now we come to the new code:
 - we look up a snap declaration that matches our current series and the given snap-id

But isn't this exactly what is done in CrossCheck (apart from the fact that we also have the assertion and can return it)?